### ISSUE TITLE
Yes, as the error says, it takes a minute for the stats to be collected. `cri_stats_provider` pulls filesystem utilization from cadvisor, which takes a bit to initialize and collect data before it can be queried.  If you look at the complete logs, you'll see another error preceding it:

### ISSUE THREAD

**Issue by adeel14553** [k3s-io/k3s#8695]
Yes, as the error says, it takes a minute for the stats to be collected. `cri_stats_provider` pulls filesystem utilization from cadvisor, which takes a bit to initialize and collect data before it can be queried.  If you look at the complete logs, you'll see another error preceding it:

```
cri_stats_provider.go:375] Failed to get the info of the filesystem with mountpoint "/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs": unable to find data in memory cache.
```

This should be harmless and clear within a few seconds after the node starts. Are you seeing otherwise?

_Originally posted by @brandond in https://github.com/k3s-io/k3s/issues/1857#issuecomment-637694918_

**Comment by adeel14553** [comment]
Yes, you are correct but somehow i get different error, i have restart node as well as k3s-agent on my worker node but still.
`Oct 19 04:31:38 nodek2 k3s[41185]: E1019 04:31:38.227900   41185 cri_stats_provider.go:455] "Failed to get the info of the filesystem with mountpoint" err="failed to get device for dir \"/var/lib/lxd/images\": stat failed on /var/lib/lxd/images with error: no such file or directory" mountpoint="/var/lib/lxd/images"`

**Comment by brandond** [comment]
Are you running k3s in a lxd container?

**Comment by adeel14553** [comment]
Yes, I have followed [this](https://github.com/automaticserver/lxe/blob/master/doc/examples/multi-node-cluster-with-k3s.md). Everything works. The container is created and then shutdown after 30 or so.

**Comment by brandond** [comment]
Ok. That's not something we've tested or would know how to support, I would suggest going to the maintainers of that project and asking for help.
